### ISSUE TITLE
loginDisabled (and other) attribute missing from inetOrgPerson

### ISSUE THREAD

**Issue by dhardiker** [osixia/docker-openldap#291]
My Google foo is failing me, so I've resorted to asking for help. Sorry if this is the wrong place!

I can get LDAP up and running just fine, I can create user which look similar to:

dn: uid=dhardiker,ou=People,dc=example,dc=org
objectClass: inetOrgPerson
objectClass: posixAccount
objectClass: shadowAccount
objectClass: ldapPublicKey
...attributes... 

Adding in uid, sn, givenName, cn, uidNumber, gidNumber, loginShell, mail, userPassword and many more attributes work just fine.

However, I want to be able to lock users out (so they can't authenticate on servers and connected systems) and to do this I think I need to use the loginDisabled attribute. https://ldapwiki.com/wiki/InetOrgPerson suggests that this (and much more) should exist, however I can't find them in the config DB (sudo docker exec openldap slapcat -n 0 | less).

Am I barking up the wrong tree? Again, sorry if all of this is me being stupid and I should be looking somewhere else. I thought I had my head around LDAP, but I guess not!

**Comment by dhardiker** [comment]
After further digging, I think this is because those login fields come from EDirectory (which is Novell Directory Services). Unfortunately I can't find a schema for that to add easily, the closest I could find is this: https://buck.ludd.ltu.se/ludd/ldap3/blob/6c45b95cc85c71d0cc5b26ebb07442a3c3f952fd/ldap3/protocol/schemas/edir888.py

I found some instructions on implementing the userAccountControl attribute (for similar Microsoft AD compatibility) and build my own auxiliary class schema which defines loginDisabled as per edir888.py - building my own schema and dropping it into the image by mapping it into /container/service/slapd/assets/config/bootstrap/schema/custom.schema seems to get it picked up, however I've had to apply to IANA to get an OID so I can make sure I don't overlap. https://ldapwiki.com/wiki/How+To+Get+Your+Own+OID

Given that SSSD seem to use this field by default to allow/deny access to a system that is authenticated using LDAP, this seems like a crazy amount of effort so I can setup a LDAP server which backs Keycloak (where Keycloak uses LDAP authoritiively without syncing), and I have services either use Keycloak (SAML / OIDC) or LDAP (readonly) directly.

I'll keep updating this as I find more information out, hopefully it'll be of use to others behind me trying to do similar and scratching their heads.

**Comment by dhardiker** [comment]
Closing the issue as it's not really related to this docker image or OpenLDAP. See my earlier comments about where to look - but I've had to implement my own attribute & class broadly following this: https://openstackcookbook.com/2015/08/17/installing-and-configuring-openldap/ (but not for the Microsoft AD userAccountControl attribute).

My custom.schema looks like this:
```
attributetype ( 2.16.840.1.113719.1.1.4.1.40
	NAME 'enabled'
	DESC 'Account enabled for login'
	SYNTAX 1.3.6.1.4.1.1466.115.121.1.7
        SINGLE-VALUE )

objectclass	( 1.2.840.113556.1.5.9
	NAME 'user'
	DESC 'a user'
	SUP top
	AUXILIARY
	MAY ( enabled ) )

```

and I use it with this: ` -v /path/to/the/custom.schema:/container/service/slapd/assets/config/bootstrap/schema/custom.schema`

I've applied for an OID from IANA which will replace  the objectclass and attributetype OIDs. I'm not sure what they'll become yet, the last piece of this particular puzzle I think.
